File: src/main/java/org/redrune/net/packet/context/impl/WalkPacketContext.java
package org.redrune.net.packet.context.impl;

import org.redrune.game.content.entity.actor.player.event.player.PlayerWalkEvent;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.net.packet.context.PacketContext;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
public class WalkPacketContext extends PacketContext {
	
	/**
	 * The destination x to travel to
	 */
	private final int destX;
	
	/**
	 * The destination y to travel to
	 */
	private final int destY;
	
	/**
	 * If the player should be forced to run
	 */
	private final boolean forceRun;
	
	public WalkPacketContext(int destX, int destY, boolean forceRun) {
		this.destX = destX;
		this.destY = destY;
		this.forceRun = forceRun;
	}
	
	@Override
	public void handle(Player player) {
		player.getEventManager().start(new PlayerWalkEvent(destX, destY, forceRun));
	}
}

File: src/main/java/org/redrune/networking/packet/context/impl/WalkPacketContext.kt
package org.redrune.networking.packet.context.impl

import org.redrune.game.content.entity.actor.player.event.player.PlayerWalkEvent
import org.redrune.game.entity.actor.player.Player
import org.redrune.networking.packet.context.PacketContext

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-04
 */
class WalkPacketContext(
    /**
     * The destination x to travel to
     */
    private val destX: Int,
    /**
     * The destination y to travel to
     */
    private val destY: Int,
    /**
     * If the player should be forced to run
     */
    private val forceRun: Boolean
) : PacketContext() {
    override fun handle(player: Player) {
        player.eventManager.start(PlayerWalkEvent(destX, destY, forceRun))
    }
}